sound/soc/sof/trace.c

```diff
@@ -44,8 +44,8 @@ static int trace_filter_parse_entry(struct snd_sof_dev *sdev, const char *line,
 	int ret;
 
 	/* ignore empty content */
-	sscanf(line, " %n", &read);
-	if (read == len)
+	ret = sscanf(line, " %n", &read);
+	if (!ret  && read == len)
```

Comment:
I thought sscanf can return EOF on an empty string or whitespaces only

Reply:
interesting, indeed, that's what `sscanf(3)` says, but looking at the kernel implementation of `vscanf()` it can never return negative numbers.

Reply:
If we wanted to be on a totally safe side here we could do `if (ret < 1 && read == len)` and BTW - there are 2 spaces between the `ret` and the `&&` ;-)

Reply:
@lyakh I think it's cleaner to handle the error case with sscanf on L51.  If ret<0 (although kernel won't return it as you note), value of read is still undefined in this case, so the whole condition is not necessarily true. 

But ugh, yes, extra whitespace :( .. I'll fix that and update.